features/nanostack/mbed_lib.json

```diff
@@ -0,0 +1,3 @@
+{
+    "name": "nanostack-upper"
```

Comment:
I'll propose "nanostack-adaptation"

Reply:
i believe that this change is in @theotherjimmy original PR

Reply:
@SeppoTakalo Actually, that's a hack that I have not yet removed. Could you comment on the appropriate folders to label such that we can completely exclude nanostack code from the build?

Reply:
Do you mean that all folders within `/features/nanostack/` should have their own mbed_lib.json files?

Reply:
Perhaps. It depends on what you would consider a complete "library", that could be turned on/off as one. If the code within the subdirectories in nanostack can be enabled independently, then it would make sense to have `mbed_lib.json` in the root so that a user could use `requires` on that library.

Reply:
![image](https://user-images.githubusercontent.com/3104794/52404621-1e380f80-2ad2-11e9-84fc-ece89bf997ea.png)

Does this image help?

There are clear dependencies that can be expressed, so maybe it makes sense to all components being separate.

`nanostack-eventloop` is actually used also with Mbed Client.

Couple of helper libraries also missing from the picture:
* features/frameworks/mbed-client-randlib (used by nanostack)
* features/frameworks/mbed-coap (used by coap-service and Pelion Client)
* features/frameworks/mbed-trace (used by Nanostack and Pelion Client, and various other libraries)
* features/frameworks/nanostack-libservice (used by pretty much all Nanostack libraries and even Pelion Client).

So.. after a longer thought.. it seems that "features/nanostack" should be empty, no .lib files there, and all that is inside it, should have their equivalent .lib files.

Reply:
@SeppoTakalo @theotherjimmy Could you move this comment to #9561?